library/CMakeLists.txt

```diff
@@ -76,11 +76,12 @@ add_library(ipfs_client
       ${sources}
       ${headers} #Mostly for IDEs' sake - this used to be considered bad practice but... generator expressions and such
   )
-if(CLANG_TIDY_EXE)
+if(USE_CLANG_TIDY AND CLANG_TIDY_EXE)
```

Comment:
This is the essence of the change that motivates the PR.

If the clang-tidy you have installed on the system is too old to handle the options in the.clang-tidy file... or not installed... the build should just continue without failing all over the place for failure to generate additional static-analysis warnings.
 
 This is most important in some CI environments, and in particular the conan-center Windows builders have ancient clang-tidy installed.